### PR TITLE
refactor(core): Migrate top level sidebar to React

### PR DESCRIPTION
**Pull request by atomrc**

### CODE REVIEW

File: src/page/template/wire-main.htm [hunk 1]
```diff
@@ -7,19 +7,20 @@
     <div id="app" class="app">
       <div id="left-column" class="left-column" data-bind="css: {'left-column--light-theme': !isActivatedAccount()}">
         <!-- ko if: webappLoaded -->
-          #include('list/background.htm')
+          <left-sidebar params="
```
Comment (atomrc) [inline src/page/template/wire-main.htm:10]: The left sidebar component now has the logic to display the right content depending on the view state, this is why `archive` and both `preferences` have disappeared from the file

File: src/script/page/left-sidebar/panels/ListWrapper.tsx [hunk 2]
```diff
@@ -76,30 +83,24 @@ const LeftListWrapper: React.FC<LeftListWrapperProps> = ({listViewModel, openSta
...
-            title={t('tooltipSearchClose')}
-            data-uie-name="do-close-preferences"
-          >
-            <Icon.Close />
-          </button>
-        </div>
-        <div css={style} ref={setScrollbarRef}>
-          {children}
-        </div>
+    <div id={id} className={`left-list-${id}`} css={style} aria-hidden={isVisible ? 'false' : 'true'}>
```
Comment (atomrc) [inline src/script/page/left-sidebar/panels/ListWrapper.tsx:86]: The `Transition` is now handled by the parent component

File: src/script/page/left-sidebar/panels/Preferences.tsx [hunk 3]
```diff
@@ -57,18 +58,12 @@ const PreferenceItem: React.FC<{
...
 };
 
-const Preferences: React.FC<PreferencesProps> = ({listViewModel, contentViewModel, isTemporaryGuest}) => {
+const Preferences: React.FC<PreferencesProps> = ({listViewModel, contentViewModel, onClose}) => {
   const {state: contentState} = useKoSubscribableChildren(contentViewModel, ['state']);
 
   const isDesktop = Runtime.isDesktopApp();
   const supportsCalling = Runtime.isSupportingLegacyCalling();
 
-  const close = () => {
```
Comment (atomrc) [inline src/script/page/left-sidebar/panels/Preferences.tsx:66]: This logic is handled by the parent component

File: src/script/page/left-sidebar/LeftSidebar.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+          listViewModel={listViewModel}
+          onClose={goHome}
+        ></Archive>
+      );
+  }
+  return (
+    <>
+      <Background selfUser={selfUser} assetRepository={assetRepository} />
+      <TransitionGroup>
+        <Animated key={state}>{content}</Animated>
```
Comment (atomrc) [inline src/script/page/left-sidebar/LeftSidebar.tsx:87]: This is now where we transition from one component to another (the slide/fade in animation is handled there)

**Review by tlebon — APPROVED** [review]

**Pull request merged**